### PR TITLE
oracle vault: Use instance principal if auth is empty

### PR DESCRIPTION
**Pull request by EladGabay**
Currently the oracle vault's secretstore uses a specific user credentials.
This commit introduce a new way to access the vault, using the instance principal.

All user's details moved to "auth" section in the OracleProvider which now is optional.
If "auth" is empty, by default, we use the instance principal, otherwise if specified user's auth details, we use them.

In addition:
- Fixed the fingerprint secret reference which until now used the privatekey secret instead of its reference.
- Bump OCI SDK version.

### CODE REVIEW

**Comment by EladGabay** [comment]
@knelasevero Hey :)
I tested it both using the instance principal, and the user auth.
Note that it's a breaking change (CRD changes), hope it's OK since it's still alpha.

**Comment by gusfcarvalho** [comment]
Hi @EladGabay ! Indeed our resources are still on v1alpha1, but we just started to move forward to v1beta1 :sweat_smile:. 

This is not mandatory, but it would be possible to rearrange it to be less disruptive to the users we have right now?

Also tagging @KianTigger

**Comment by EladGabay** [comment]
Hi @gusfcarvalho ! :)
Note that it's much more making sense now, no mixing auth params outside of the auth section.
For example, the tenancy OCID is not the vault OCID as mentioned now, it's the user's tenancy.
(that means the user might be in tenancy A and vault in tenancy B, just need permissions).
So I'd suggest to proceed with the changes now, before it's late :)

File: e2e/suite/oracle/provider.go [hunk 1]
```diff
@@ -113,7 +113,11 @@ func (p *oracleProvider) BeforeEach() {
 		Spec: esv1alpha1.SecretStoreSpec{
 			Provider: &esv1alpha1.SecretStoreProvider{
 				Oracle: &esv1alpha1.OracleProvider{
-					Auth: esv1alpha1.OracleAuth{
+					Region: p.region,
+					Vault:  "vaultOCID",
```
Comment (gusfcarvalho) [inline e2e/suite/oracle/provider.go:117]: Is "vaultOCID" the default value for this field? If it is, maybe we should consider this field as optional and set the default.

**Comment by gusfcarvalho** [comment]
Hi @EladGabay ! 
Our best indication of active users are through the analytics of the docs site. Oracle Vault provider had +200 page search for the last month, so I would expect to have some amount of active users.

I did talk to @knelasevero yesterday, and, although I personally would still recommend searching for a non breaking-change approach (since the CRD will change, any old CRD applied will have to be deleted before installing a new one), we are incorporating your changes to v1alpha1.

File: e2e/suite/oracle/provider.go (hunk 1 above)
Comment (EladGabay) [reply e2e/suite/oracle/provider.go:117]: Actually I don't know if this e2e is even running somewhere, currently (before my change) it seems that the CRD has no vault at all which is not a valid CRD.

Comment (gusfcarvalho) [reply e2e/suite/oracle/provider.go:117]: @knelasevero do you know if this e2e is actually being used somewhere?

Comment (knelasevero) [reply e2e/suite/oracle/provider.go:117]: Implemented, but never called. We don't have a dedicated oracle account to use run them against it

**Comment by EladGabay** [comment]
Let's merge :tada: ?

**Comment by moolen** [comment]
Fine for me. We gotta make sure to provide warnings in the changelog when releasing this.

**Pull request merged**